### PR TITLE
Andromeda Keyboard

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding the Andromeda Keyboard to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ai03/andromeda/mcuconf.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/cannonkeys/tsukuyomi/mcuconf.h -r platforms/chibios/GENERIC_STM32_F072XB/configs/mcuconf.h`
+ */
+
```
Comment (tzarc) [inline keyboards/ai03/andromeda/mcuconf.h:21]:
```suggestion
```

File: keyboards/ai03/andromeda/halconf.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/cannonkeys/tsukuyomi/halconf.h -r platforms/chibios/common/configs/halconf.h`
+ */
+
```
Comment (tzarc) [inline keyboards/ai03/andromeda/halconf.h:21]:
```suggestion
```

File: keyboards/ai03/andromeda/config.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xA103
+#define PRODUCT_ID      0x000A
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    ai03 Design Studio
+#define PRODUCT         Andromeda
+#define DESCRIPTION     Super High End TKL
```
Comment (tzarc) [inline keyboards/ai03/andromeda/config.h:26]:
```suggestion
```

File: keyboards/ai03/andromeda/config.h [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
```
Comment (tzarc) [inline keyboards/ai03/andromeda/config.h:62]:
```suggestion
```

File: keyboards/ai03/andromeda/chconf.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/cannonkeys/tsukuyomi/chconf.h -r platforms/chibios/common/configs/chconf.h`
+ */
```
Comment (tzarc) [inline keyboards/ai03/andromeda/chconf.h:20]:
```suggestion
```

File: keyboards/ai03/andromeda/chconf.h [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+#pragma once
+
+#define CH_CFG_ST_FREQUENCY 10000
+
+#define CH_CFG_OPTIMIZE_SPEED FALSE
+
+#define CH_CFG_USE_CONDVARS_TIMEOUT FALSE
+
```
Comment (tzarc) [inline keyboards/ai03/andromeda/chconf.h:29]: Are these needed?

File: keyboards/ai03/andromeda/halconf.h [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+#pragma once
+
+#define HAL_USE_PWM TRUE
+
+#define HAL_USE_SPI TRUE
+
```
Comment (tzarc) [inline keyboards/ai03/andromeda/halconf.h:27]: Are these needed? If not, this file can be removed.

File: keyboards/ai03/andromeda/chconf.h (hunk 6 above)
Comment (tzarc) [reply keyboards/ai03/andromeda/chconf.h:29]: Can try removing this file in general.

File: keyboards/ai03/andromeda/mcuconf.h [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+#pragma once
+
+#include_next <mcuconf.h>
+
+#undef STM32_PWM_USE_TIM3
+#define STM32_PWM_USE_TIM3 TRUE
+
+#undef STM32_SPI_USE_SPI2
+#define STM32_SPI_USE_SPI2 TRUE
```
Comment (tzarc) [inline keyboards/ai03/andromeda/mcuconf.h:30]: If these aren't required, remove this file.

**Review by tzarc — CHANGES_REQUESTED** [review]

File: keyboards/ai03/andromeda/chconf.h (hunk 6 above)
Comment (awkannan) [reply keyboards/ai03/andromeda/chconf.h:29]: They aren't necessary and I will remove

File: keyboards/ai03/andromeda/halconf.h (hunk 7 above)
Comment (awkannan) [reply keyboards/ai03/andromeda/halconf.h:27]: iirc, andromeda uses neither SPI nor PWM so I can safely remove it

File: keyboards/ai03/andromeda/mcuconf.h (hunk 8 above)
Comment (awkannan) [reply keyboards/ai03/andromeda/mcuconf.h:30]: 👍

**Review by tzarc — APPROVED** [review]

File: keyboards/ai03/andromeda/rules.mk [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no      # Custom matrix file
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (fauxpark) [inline keyboards/ai03/andromeda/rules.mk:16]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ai03/andromeda/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+enum layer_names {
+    _BASE,
+    _FN1
```
Comment (fauxpark) [inline keyboards/ai03/andromeda/keymaps/default/keymap.c:20]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ai03/andromeda/readme.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+# ai03 Andromeda
+
+* Keyboard Maintainer: [ai03](https://github.com/ai03-2725)
+* Hardware Supported: Andromeda PCB
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ai03/andromeda:default
+
```
Comment (noroadsleft) [inline keyboards/ai03/andromeda/readme.md:9]: Could you please add instructions here on how to reset the keyboard into bootloader mode?

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/12676/files).

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**